claimer/src/check-l2-status.ts

```diff
@@ -0,0 +1,350 @@
+
+export interface L2Status {
+  isUp: boolean
+  lastProofBlock?: number
+  lastProofTimestamp?: number
+  timeSinceLastProof?: number
+  error?: string
+}
+
+// get all L2/L3 chains
+export async function getL2s() {
+  const l2Chains = Object.values(networks).filter(
+    (chain): chain is ChildNetworkConfig =>
+      'stack' in chain &&
+      'parentChainId' in chain &&
+      chain.chainId !== chain.parentChainId
+  )
+
+  return l2Chains
+}
+
+// Check status for each L2/L3 chain
+export async function checkL2Chains() {
+  const l2Chains = await getL2s()
+  for (const chain of l2Chains) {
+    try {
+      const status = await checkL2Status(Number(chain.chainId))
+      console.log(`\nL2 Status Results for ${chain.name} ${chain.chainId}:`)
+      console.log('------------------')
+      console.log(`Chain is up: ${status.isUp}`)
+
+      if (!status.isUp) {
+        console.log(`FAIL: ${status.error}`)
+      }
+
+      if (status.lastProofBlock) {
+        console.log(`Last proof block: ${status.lastProofBlock}`)
+      }
+
+      if (status.lastProofTimestamp) {
+        console.log(
+          `Last proof timestamp: ${new Date(status.lastProofTimestamp * 1000).toISOString()}`
+        )
+      }
+
+      if (status.timeSinceLastProof) {
+        console.log(
+          `Time since last proof: ${status.timeSinceLastProof} seconds`
+        )
+      }
+
+      if (status.error) {
+        console.log(`Error: ${status.error}`)
+      }
+    } catch (error) {
+      console.log(error)
+    }
+  }
+}
+
+export async function checkL2Status(chainId: number): Promise<L2Status> {
+  try {
+    const chain = networks[chainId] as ChildNetworkConfig
+
+    if (!chain) {
+      throw new Error(`Chain ${chainId} not found in config`)
+    }
+
+    // Get L1 provider for checking proofs
+    const l1Provider = new JsonRpcProvider(networks[chain.parentChainId].rpc[0])
+
+    // Get current L1 block
+    const currentL1Block = await l1Provider.getBlockNumber()
+
+    // Different handling based on L2 stack type
+    if (chain.stack === 'optimism') {
+      return await checkOptimismStatus(chain, l1Provider, currentL1Block)
+      // } else if (chain.stack === 'optimism-alt') {
+      //   return await checkOptimismBedrockStatus(chain, l1Provider, currentL1Block)
+      // } else if (chain.stack === 'arbitrum') {
+      //   return await checkArbitrumStatus(chain, l1Provider, currentL1Block)
+      // } else if (chain.stack === 'zksync') {
+      //   return await checkZkSyncStatus(chain, l1Provider, currentL1Block)
+    } else {
+      throw new Error(`Unsupported L2 stack type: ${chain.stack}`)
+    }
+  } catch (error) {
+    return {
+      isUp: false,
+      error: error instanceof Error ? error.message : 'Unknown error occurred',
+    }
+  }
+}
+
+// deprec version of OP stack using l2OutputOracle (before bedrock)
+// https://gov.optimism.io/t/final-protocol-upgrade-7-fault-proofs/8161
+async function checkOptimismStatus(
```

Comment:
Let's move that to a dedicated op file since we will need to do that for all stacks!